cmd/common.go

```diff
@@ -344,92 +346,157 @@ func GetResourceType(c *cliclient.MasterClient, resource string) (string, error)
 	var byName *ntypes.Resource
 
 	for _, schemaType := range types {
-		rt, err := GetResourceType(c, schemaType)
+		schemaClient, err := lookupSchemaClient(c, schemaType)
 		if err != nil {
-			logrus.Debugf("Error GetResourceType: %v", err)
 			return nil, err
 		}
-		var schemaClient clientbase.APIBaseClientInterface
-		// the schemaType dictates which client we need to use
-		if c.CAPIClient != nil {
-			if strings.HasPrefix(rt, "cluster.x-k8s.io") {
-				schemaClient = c.CAPIClient
-			}
-		}
-		if c.ManagementClient != nil {
-			if _, ok := c.ManagementClient.APIBaseClient.Types[rt]; ok {
-				schemaClient = c.ManagementClient
-			}
-		}
-		if c.ProjectClient != nil {
-			if _, ok := c.ProjectClient.APIBaseClient.Types[rt]; ok {
-				schemaClient = c.ProjectClient
-			}
-		}
-		if c.ClusterClient != nil {
-			if _, ok := c.ClusterClient.APIBaseClient.Types[rt]; ok {
-				schemaClient = c.ClusterClient
-			}
-		}
-
 		// Attempt to get the resource by ID
-		var resource ntypes.Resource
-
-		if err := schemaClient.ByID(schemaType, name, &resource); !clientbase.IsNotFound(err) && err != nil {
-			logrus.Debugf("Error schemaClient.ByID: %v", err)
+		resource, err := lookupByIDWithClient(schemaClient, name, schemaType)
+		if err != nil && !errors.Is(err, errNotFound) {
 			return nil, err
-		} else if err == nil && resource.ID == name {
-			return &resource, nil
 		}
-
-		// Resource was not found assuming the ID, check if it's the name of a resource
-		var collection ntypes.ResourceCollection
-
-		listOpts := &ntypes.ListOpts{
-			Filters: map[string]interface{}{
-				"name":         name,
-				"removed_null": 1,
-			},
+		if resource != nil {
+			return resource, nil
 		}
 
-		if err := schemaClient.List(schemaType, listOpts, &collection); !clientbase.IsNotFound(err) && err != nil {
-			logrus.Debugf("Error schemaClient.List: %v", err)
+		// Resource was not found assuming the ID, check if it's the name of a resource
+		resource, err = lookupByNameWithClient(schemaClient, name, schemaType)
+		if err != nil && !errors.Is(err, errNotFound) {
 			return nil, err
 		}
 
-		if len(collection.Data) > 1 {
-			ids := []string{}
-			for _, data := range collection.Data {
-				ids = append(ids, data.ID)
-			}
-			return nil, fmt.Errorf("Multiple resources of type %s found for name %s: %v", schemaType, name, ids)
-		}
-
 		// No matches for this schemaType, try the next one
-		if len(collection.Data) == 0 {
+		if resource == nil {
 			continue
 		}
 
 		if byName != nil {
-			return nil, fmt.Errorf("Multiple resources named %s: %s:%s, %s:%s", name, collection.Data[0].Type,
-				collection.Data[0].ID, byName.Type, byName.ID)
+			return nil, fmt.Errorf("multiple resources named %s: %s:%s, %s:%s", name, resource.Type,
+				resource.ID, byName.Type, byName.ID)
 		}
 
-		byName = &collection.Data[0]
+		byName = resource
 
 	}
 
 	if byName == nil {
-		return nil, fmt.Errorf("Not found: %s", name)
+		return nil, fmt.Errorf("%w: %s", errNotFound, name)
```

Comment:
```suggestion
		return nil, fmt.Errorf("%s: %w", name, newRole)
```
Wrapping convention is to have the wrapped error at the end prefixed by a colon.